src/main/java/Duke.java

```diff
@@ -1,10 +1,100 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    public static void inputChecker() {
+        Scanner in = new Scanner(System.in); // Declared a scanner object
+        String line; // Declared a string object to take in user input
+        Task[] list; // Declared a list with string data type
+        list = new Task[100]; // created a list of 100 elements
+        int counter = 0;
+        int index;
+        Task t;
+
+        // Start of user input
+        while (true) {
+            line = in.nextLine(); // Takes in user input
+
+            boolean mark = line.trim().toLowerCase().startsWith("mark");
+            boolean unmark = line.trim().toLowerCase().startsWith("unmark");
+
+            // User wants to exit
+            if (line.equals("bye")) {
+                System.out.println("Bye human. Come back soon !");
+                break;
+            }
+
+            // User wants to display the list of tasks
+            else if (line.equalsIgnoreCase("list")) {
+                for (Task task : list) {
+                    if (task == null){
```

Comment:
Efficient way to end the loop when null is encountered. Ensure tasks are packed with no 'null' gaps